### PR TITLE
Added tests for calculateTransitionSpeed

### LINKED ISSUE #1052: Investigate and resolve possibility of incorrect transition speed if a path doubles back on itself
### Description of the task
Investigate and resolve possibility of incorrect transition speed if a path doubles back on itself: https://github.com/UBC-Thunderbots/Software/pull/1043#discussion_r347594658

For context, calculateTransitionSpeedBetweenSegments in navigator/navigator.cpp is used to slow down the robot depending on how dramatically the direction changes from one segment to the next. @EvanMorcom pointed out that the project used could return a negative value, which we would then take the length of, if a path doubled back on itself. In the case of two segments that go in opposite directions, we would want the final speed at the end of the first segment to be zero to facilitate changing directions. The length of the negative projection could be nonzero and possibly quite large. While paths generally don't double back on themselves, using a this transition speed would effectively make the robot go too fast past the transition point and not change direction at the right location. That would be a bug.

### Acceptance criteria

- [ ] Create a situation that has negative transition speed, which we then take the length of
- [ ] Find a reasonable way to deal with the situation (set transition speed to 0?)
- [ ] Add regression tests for this situation

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Add tests for calculateTransitionSpeed to check for a path that double back on itself

### Testing Done
added a test case

### Resolved Issues
resolves #1052

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by MathewMacDougall — DISMISSED** [review]

File: src/software/ai/navigator/navigator_test.cpp [hunk 1]
```diff
@@ -365,6 +365,28 @@ TEST_F(NoPathNavigatorTest,
                                       testp1, testp2, testp3, final_speed));
 }
 
+TEST_F(NoPathNavigatorTest,
+       calculateTransitionSpeedBetweenSegments_tests_opposite_segments)
+{
+    Point testp1, testp2, testp3;
+    double final_speed;
+    // case 1
```
Comment (kelvinkoon) [inline src/software/ai/navigator/navigator_test.cpp]: Commenting `case 1` and `case 2` feels very vague, particularly when both 'cases' are under the same test name.

**Review by kelvinkoon — COMMENTED** [review]
LGTM besides 1 nit.

File: src/software/ai/navigator/navigator_test.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/software/ai/navigator/navigator_test.cpp]: Do you want the comments removed?

Comment (kelvinkoon) [reply src/software/ai/navigator/navigator_test.cpp]: I'd prefer them to be expanded upon what the 1st and 2nd cases are if possible.

Comment (jonathanlew) [reply src/software/ai/navigator/navigator_test.cpp]: They are two very similar cases that test opposite segments

Comment (garethellis0) [reply src/software/ai/navigator/navigator_test.cpp]: Perhaps we could just leave a comment indicating the difference / relation between the two cases to make this clear?

Comment (jonathanlew) [reply src/software/ai/navigator/navigator_test.cpp]: ok done

**Review by MathewMacDougall — APPROVED** [review]

**Review by garethellis0 — APPROVED** [review]

**Review by kelvinkoon — APPROVED** [review]
LGTM! :tada:

**Pull request merged**